### PR TITLE
chore: added parallel runs to the framework, .env support, extract check OS improve assertions

### PR DESCRIPTION
**Pull request by khru**
Peek 2023-09-08 20-31

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1 @@
+PARALLEL_RUN=
```
Comment (Chemaclass) [inline .env.example:1]: isn't the `.env.dist` the actual example file already? Why do we need this `.env.example` file?

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,10 @@
 - Added `assertNotMatches`
 - Added `make test/watch` to run your test every second
 - Added time taken to run the test in ms (only to Linux)
+- Simplified assertions over test results
+- Added pre-commit to the project
+- Allow parallel tests to run base on a .env configuration enabled by default
+- I
```
Comment (Chemaclass) [inline CHANGELOG.md:12]: `- I` typo?

File: Makefile [hunk 3]
```diff
@@ -34,20 +36,24 @@ help:
 	@echo "usage: make COMMAND"
 	@echo ""
 	@echo "Commands:"
-	@echo "  list-tests               List all the test under the tests directory"
+	@echo "  tests/list               List all the test under the tests directory"
```
Comment (Chemaclass) [inline Makefile:39]: What do you think about keeping this as `test/list`?

**Review by JesusValera — COMMENTED** [review]
Agree with @Chemaclass suggestions, the rest is ok 💯

File: .env.example (hunk 1 above)
Comment (khru) [reply .env.example:1]: > isn't the .env.dist the actual example file already?

I will deleted it's not necesary
https://github.com/Chemaclass/bashunit/blob/4c6aa107d87d52937d6bc10a3ae545253db9dcda/bashunit#L9

File: CHANGELOG.md (hunk 2 above)
Comment (khru) [reply CHANGELOG.md]: Done

File: .env.example (hunk 1 above)
Comment (khru) [reply .env.example:1]: Done

File: Makefile (hunk 3 above)
Comment (khru) [reply Makefile:39]: Done

Comment (Chemaclass) [reply Makefile:39]: You forgot to update the help documentation:)

**Review by Chemaclass — APPROVED** [review]

**Pull request merged**